pkg/cloudprovider/cloud.go

```diff
@@ -58,11 +58,10 @@ func GetLoadBalancerName(service *api.Service) string {
 
 // TCPLoadBalancer is an abstract, pluggable interface for TCP load balancers.
 type TCPLoadBalancer interface {
-	// TCPLoadBalancerExists returns whether the specified load balancer exists.
 	// TODO: Break this up into different interfaces (LB, etc) when we have more than one type of service
-	// TODO: This should really return the details of the load balancer so we can
-	// determine if it matches the needs of a service rather than if it exists.
-	TCPLoadBalancerExists(name, region string) (bool, error)
+	// GetTCPLoadBalancer returns whether the specified load balancer exists, and
+	// if so, what its IP address or hostname is.
```

Comment:
I don't like the "or" part of this.  Can we make this a struct that has both hostname and ip and fill in one (or both) as appropriate?  Also make the IP address a net.IPAddress.

We've already had pain from hostname vs. ip confusion in other part so of the stack that we had to iron out.